### PR TITLE
✨ feat(Cryptography): Post-Quantum cryptography

### PR DESCRIPTION
**Pull request by raxhvl**
## Wiki PR Checklist

- [x] Cover the impact of quantum computing on cryptography.
- [x] Cover quantum algorithms being reviewed by NIST.
- [x] Review changes to ensure there are no typos, see instructions bellow

### CODE REVIEW

**Comment by raxhvl** [comment]
Depends on #147

File: docs/wiki/Cryptography/post-quantum-cryptography.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Post-Quantum Cryptography
+
+Classical cryptography safeguards information by leveraging the inherent difficulty of certain mathematical problems. These problems fall under the area of mathematical research called the ["Hidden Subgroup Problem (HSP)"](https://en.wikipedia.org/wiki/Hidden_subgroup_problem). Imagine a large group with a secret subgroup known only to insiders, these problems makes determining the structure of the secret subgroup (size, elements) computationally intractable for an outsider. Whereas, someone with the "secret" (the private key) can easily identify the subgroup.
+
+Public-key cryptography leverages this concept. Algorithms like RSA, DSA, and [ECDSA](/wiki/Cryptography/ecdsa.md) rely on hidden subgroup problems like prime factorization of large integers or discrete logarithm calculations to secure private keys. The difficulty of solving these problems increases exponentially with key size, making brute-force attacks impractical for classical computers. This inherent difficulty safeguards encrypted data.
```
Comment (taxmeifyoucan) [inline docs/wiki/Cryptography/post-quantum-cryptography.md:5]: I would make a bit more clear here - ecc relies on discrete logarithm problem and RSA on integer factorization. Especially in context of Ethereum, I would highlight the former. And that both are instances of the hidden subgroup problem, it's implied here but maybe not as clear for a new reader

File: docs/wiki/Cryptography/post-quantum-cryptography.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+However, the landscape is shifting.
+
+Quantum computers, harnessing the principles of quantum mechanics, offer novel computational approaches. Certain quantum algorithms can solve these classical cryptographic problems with exponential efficiency compared to their classical counterparts. This newfound capability poses a significant threat to the security of data encrypted with classical cryptography.
```
Comment (taxmeifyoucan) [inline docs/wiki/Cryptography/post-quantum-cryptography.md:9]: I would elaborate on this part a little bit. Maybe mentioning qubits and that current quantum computers don't have enough of them to break crypto but there are threats like HNDL https://wikipedia.org/wiki/Harvest_now,_decrypt_later.

File: docs/wiki/Cryptography/post-quantum-cryptography.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+This is where the field of post-quantum cryptography comes in. It aims to develop new algorithms that remain secure even in the presence of powerful quantum computers.
+
+Post-quantum cryptography is an active area of research. Currently, NIST is evaluating submissions to standardize quantum-resistant algorithms.
+
+## Selected Algorithms 2022
+
+### Public-key Encryption and Key-establishment Algorithms
+
+- [CRYSTALS-KYBER](https://pq-crystals.org/) by Peter Schwabe et al.
```
Comment (taxmeifyoucan) [inline docs/wiki/Cryptography/post-quantum-cryptography.md:21]: awesome! Signal switched to using crystals-kyber in addition to ecc, maybe worth mentioning somwhere it's being used in production?

File: docs/wiki/Cryptography/post-quantum-cryptography.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+However, the landscape is shifting.
+
+Quantum computers, harnessing the principles of quantum mechanics, offer novel computational approaches. Certain quantum algorithms can solve these classical cryptographic problems with exponential efficiency compared to their classical counterparts. This newfound capability poses a significant threat to the security of data encrypted with classical cryptography.
+
+[Shor's algorithm](https://ieeexplore.ieee.org/document/365700) for integer factorization is the most celebrated application of quantum computing. It factors n-digit integers in a time complexity less than $O(n^3)$, a significant improvement over the best classical algorithms.
+
+This is where the field of post-quantum cryptography comes in. It aims to develop new algorithms that remain secure even in the presence of powerful quantum computers.
+
+Post-quantum cryptography is an active area of research. Currently, NIST is evaluating submissions to standardize quantum-resistant algorithms.
+
```
Comment (taxmeifyoucan) [inline docs/wiki/Cryptography/post-quantum-cryptography.md:16]: One section potentially missing here is analysis of Ethereum stack and whats post quantum upgrades are necessary. E.g. hashes are safe, accounts need a new sig scheme, kzg commitments must be upgraded, etc

**Review by taxmeifyoucan — COMMENTED** [review]
Thanks a lot Rahul! Great to have this covered on wiki

File: docs/wiki/Cryptography/post-quantum-cryptography.md (hunk 3 above)
Comment (raxhvl) [reply docs/wiki/Cryptography/post-quantum-cryptography.md:27]: Interesting. Do you have a link?

Comment (taxmeifyoucan) [reply docs/wiki/Cryptography/post-quantum-cryptography.md:27]: https://signal.org/docs/specifications/pqxdh/

File: docs/wiki/Cryptography/post-quantum-cryptography.md (hunk 1 above)
Comment (raxhvl) [reply docs/wiki/Cryptography/post-quantum-cryptography.md:5]: Reworded for more clarity.

**Comment by taxmeifyoucan** [comment]
Thank you so much for another great piece of content Rahul! I gave it another read and it looks great, merging :)

**Pull request merged**